L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryV9Imp1.cc

```diff
@@ -0,0 +1,1100 @@
+#include <fstream>
+#include <regex>
+
+
+class HGCalTriggerGeometryV9Imp1 : public HGCalTriggerGeometryBase
+{
+    public:
+        HGCalTriggerGeometryV9Imp1(const edm::ParameterSet& conf);
+
+        void initialize(const edm::ESHandle<CaloGeometry>& ) final;
+        void initialize(const edm::ESHandle<HGCalGeometry>&,
+                const edm::ESHandle<HGCalGeometry>&,
+                const edm::ESHandle<HGCalGeometry>&) final;
+        void reset() final;
+
+        unsigned getTriggerCellFromCell( const unsigned ) const final;
+        unsigned getModuleFromCell( const unsigned ) const final;
+        unsigned getModuleFromTriggerCell( const unsigned ) const final;
+
+        geom_set getCellsFromTriggerCell( const unsigned ) const final;
+        geom_set getCellsFromModule( const unsigned ) const final;
+        geom_set getTriggerCellsFromModule( const unsigned ) const final;
+
+        geom_ordered_set getOrderedCellsFromModule( const unsigned ) const final;
+        geom_ordered_set getOrderedTriggerCellsFromModule( const unsigned ) const final;
+
+        geom_set getNeighborsFromTriggerCell( const unsigned ) const final;
+
+        GlobalPoint getTriggerCellPosition(const unsigned ) const final;
+        GlobalPoint getModulePosition(const unsigned ) const final;
+
+        bool validTriggerCell( const unsigned ) const final;
+        bool disconnectedModule(const unsigned) const final;
+        unsigned triggerLayer(const unsigned) const final;
+
+    private:
+        edm::FileInPath l1tCellsMapping_;
+        edm::FileInPath l1tCellsSciMapping_;
+        edm::FileInPath l1tWafersMapping_;
+        edm::FileInPath l1tModulesMapping_;
+        edm::FileInPath l1tCellNeighborsMapping_;
+        edm::FileInPath l1tCellNeighborsSciMapping_;
+
+        // module related maps
+        std::unordered_map<unsigned, unsigned> wafer_to_module_;
+        std::unordered_multimap<unsigned, unsigned> module_to_wafers_;
+
+        // trigger cell related maps
+        std::unordered_map<unsigned, unsigned> cells_to_trigger_cells_;
+        std::unordered_multimap<unsigned, unsigned> trigger_cells_to_cells_;
+        std::unordered_map<unsigned, unsigned> cells_to_trigger_cells_sci_;
+        std::unordered_multimap<unsigned, unsigned> trigger_cells_to_cells_sci_;
+        std::unordered_map<unsigned, unsigned short> number_trigger_cells_in_wafers_; 
+        std::unordered_map<unsigned, unsigned short> number_trigger_cells_in_wafers_sci_; 
+        std::unordered_map<unsigned, unsigned> wafers_to_wafers_old_;
+        std::unordered_map<unsigned, unsigned> wafers_old_to_wafers_;
+        std::unordered_set<unsigned> invalid_triggercells_;
+
+        // neighbor related maps
+        // trigger cell neighbors:
+        // - The key includes the module and trigger cell id
+        // - The value is a set of (module_id, trigger_cell_id)
+        std::unordered_map<int, std::set<std::pair<short,short>>> trigger_cell_neighbors_;
```

Comment:
It could be interesting to see if an `unordered_set` performs better than the sorted `set` here (though I assume the size of the set is quite small).

Also, it might be worth `typedef`ing `std::unordered_map<int, std::set<std::pair<short,short>>>` to something shorter/clearer, since it is used in a number of places.

Reply:
The set is just storing the neighbours, so less than 10 elements. I suppose that the difference in terms of performance will be small.
Ok for the `typedef`.